pkg/steve/pkg/devicetemplateapi/store.go

```diff
@@ -99,39 +96,19 @@ func ValidateTemplateRequest(spec v1alpha1.DeviceTemplateSpec) error {
 	if spec.DeviceResource == "" {
 		return errors.New("deviceResource is required of DeviceTemplate")
 	}
-	if spec.TemplateSpec == nil {
-		return errors.New("templateSpec is required of DeviceTemplate")
+	if spec.DisplayName == "" {
```

Comment:
> don't think we need the display name since metadata.name serves the same purpose.

Since the metadata name has a fixed format restriction, it may not be flexible enough to meet the user's naming conventions. like name v1-20200610 is not allowed